### ISSUE TITLE
Max cluster radius can't be set dynamically

### ISSUE THREAD

**Issue by toto6038** [veitbjarsch/vue-leaflet-markercluster#22]
Hello, thanks for your amazing work. As described on [leaflet-markercluster's documentation](https://github.com/Leaflet/Leaflet.markercluster?tab=readme-ov-file#options), the `maxClusterRadius` option can take a function that returns the radius depending on zoom level. However, when the `maxClusterRadius` value changes, it causes  `Error: Max cluster radius can't be set dynamically`. I wonder if this is a restriction from the upstream package (`Leaflet/Leaflet.markercluster`), or an issue that can be addressed in your vue adaptor?

**Comment by veitbjarsch** [comment]
Hey, thanks for using the package. You are correct, currently the vue adaptor expects a number as the value for `maxClusterRadius`. Nevertheless this should cause a different error (normally just a warning and not even an error). So there might be something else which goes wrong here. Can you please create a minimal reproduction project in a tool like `codesandbox` and post the link here. Then I can investigate the issue.

**Comment by toto6038** [comment]
Hi, I've made a reproduction. Please check it out: https://stackblitz.com/edit/vitejs-vite-c4utvk?file=src%2FApp.vue . You can see that when the `maxClusterRadius` option is bounded to a function that returns the radius based on zoom level, Javascript console logs `Error: Max cluster radius can't be set dynamically`.

**Comment by veitbjarsch** [comment]
Hey, @toto6038 thanks for the reproduction link. I checked it and can tell you that you experience 3 problems in this code piece.

1. You are defining the function as anonymous function directly in the template. This means, in every step when the template is reevaluated a new anonymous function is created. Because the new one points to a different place in the memory the two function (first and second template evaluation) are not the same. This results in a dynamically reassignment of the property, which leads to the: `Error: Max cluster radius can't be set dynamically`.
An easy fix here is instead of using an anonymous function, just create this function in the js setup part and pass the function by reference. This will get you rid of the error message. (It won't fix our problem... yet)

2. The second problem is the assumption that the marker cluster will dynamically calculate the clusters. The leaflet marker cluster plugin won't calculate the cluster when you zoom in or out. It calculates all possible clusters for all zoom levels in advance during the initial rendering. Using `() => (zoom>15) ? 40 : 80"` the reference will always check against the same zoom level. The level which is initially set. If you use instead the passed prop `(zoom) => (zoom>15) ? 40 : 80`. The function will work as expected.

3. And there is a third problem which is caused by this plugin. Currently The warning `Invalid prop: type check failed for prop "maxClusterRadius". Expected Number with value NaN, got Function? is shown. This warning won't stop the functionality but is indeed annoying. I going to fix this one in the upcoming with and will release a new version.

To sum it up: changing: 
 ```
<LMarkerClusterGroup :maxClusterRadius="()=>(zoom>15)?40:80" />
```` 
to
```
const someFunction = (zoom) => (zoom > 15) ? 40 : 80
---
<LMarkerClusterGroup :maxClusterRadius="someFunction" />

```
will fix your problems. The warning will still exists, but will be fixed with the next version of this plugin.

Hope I could help you.

Best Regards
Veit

**Comment by toto6038** [comment]
Thank you for your detailed explanation. It helps a lot!

Cheers

**Comment by veitbjarsch** [comment]
@toto6038 Warning should now be fixed in the latest version
